### ISSUE TITLE
KDE5 Doesn't Exist

### ISSUE THREAD

**Issue by grahamc** [NixOS/nixpkgs#23041]
Apparently there is no such thing as KDE5, it is now

 - Frameworks
 - Plasma
 - Applications

each versioned separately. KDE is the community.

I was talking on IRC (#kde on freenode) and was asked to have the references to KDE4 and KDE5 fixed, as there is no KDE5 and it has not been called KDE4 in some time.

tosky provided these links:

 - https://dot.kde.org/2014/06/26/where-kde-going-part-1
 - https://dot.kde.org/2014/07/02/where-kde-going-part-2
 - http://tsdgeos.blogspot.cz/2014/08/kde-releases-in-future.html

and offered the following resources to help distros / packagers:

 - #kde-devel on freenode
 - this ML for distributions: https://mail.kde.org/mailman/listinfo/distributions
 - this ML for packagers and packaging issues: https://mail.kde.org/mailman/listinfo/kde-distro-packagers

and noted that we can register for pre-releases for Frameworks, Plasma, and Applications for pre-testing by filing a "sysadmin" ticket here: https://phabricator.kde.org/maniphest/task/edit/form/2/ (requires registration: https://identity.kde.org/index.php?r=registration/index) and providing evidence that you're coming from NixOS.

When we have a plan, we might want to stop by #kde-devel on Freenode and have them check it out.

cc @ttuegel

**Comment by vcunat** [comment]
I think it's fine to keep having the packages grouped in `kde5.*` and called that, even if the project isn't so monolithic anymore...

**Comment by tosky** [comment]
Can't they just be kde.* ?

**Comment by vcunat** [comment]
There might be such alias, but typically we find that when such a major change happens (python2 -> python3, gtk2 -> gtk3, ...) the users of the name have to decide explicitly when to switch anyway.

**Comment by tosky** [comment]
but there is no "kde5", and it would be really confusing to move everything in that basket. Having the support from distributions is crucial. Also, from a technical point of view,KDE Applications has still kdelibs4- and Frameworks-based applications for the next 9 months. So I would suggest to really use a unified kde.* namespace if you want to go that way. Otherwise please use kdeframeworks5, kdeplasma5.*, etc.

**Comment by ttuegel** [comment]
> but there is no "kde5", and it would be really confusing to move everything in that basket. Having the support from distributions is crucial. Also, from a technical point of view,KDE Applications has still kdelibs4- and Frameworks-based applications for the next 9 months. So I would suggest to really use a unified kde.* namespace if you want to go that way. Otherwise please use kdeframeworks5, kdeplasma5.*, etc.

Currently we have `kde5.frameworks`, `kde5.applications`, and `kde5.plasma`. I'm sorry if this is not clear enough. I did this at the behest of our benevolent overlord, @edolstra. If this were a trademark issue, on the other hand...

**Comment by ttuegel** [comment]
> and offered the following resources to help distros / packagers:
> 
>     #kde-devel on freenode
>     this ML for distributions: https://mail.kde.org/mailman/listinfo/distributions
>     this ML for packagers and packaging issues: https://mail.kde.org/mailman/listinfo/kde-distro-packagers
> 
> and noted that we can register for pre-releases for Frameworks, Plasma, and Applications for pre-testing by filing a "sysadmin" ticket here: https://phabricator.kde.org/maniphest/task/edit/form/2/ (requires registration: https://identity.kde.org/index.php?r=registration/index) and providing evidence that you're coming from NixOS.

I'm subscribed to those mailing lists for... well, however long I've been packaging KDE 5 for NixOS.

Since we are a build-from-source distribution, I didn't see any reason to apply for pre-release access. If we wanted to, it would be simpler to `fetchgit` by tag anyway. 😜

As far as pre-release testing goes, the amount of time it usually takes me to patch the crap out of upstream's release is usually much longer than the pre-release window anyway, so unless I start tracking upstream Git during the development cycle, I don't see the point.

That's why we don't have Plasma 5.9, by the way: I can't get the desktop to load despite working on it since release day! But upgrading would be counterproductive anyway, since we want the Plasma 5.8 LTS release in *our* LTS release next month.

@grahamc Do you feel differently?

**Comment by tosky** [comment]
Packages and git are different: you don't have translations apart from Frameworks.

Also, for naming, please please really switch to the non-versioned namespace if you use kde. People are confused by that and try to find a non-existing kde5. kde.frameworks5, kde.plasma5 and kde.applications (unversioned, it's mixed content) is fine though; same for the rest of applications ("KDE Applications" is a well defined released product, which does not include stuff like Kile, KMyMoney etc, which have their own independent schedule; I hope that they are handled in a separate namespace).

**Comment by ttuegel** [comment]
> Packages and git are different: you don't have translations apart from Frameworks.

That is irrelevant to our testing process.

> Also, for naming, please please really switch to the non-versioned namespace if you use kde.

So I understand correctly, would you be happy if we just replaced every occurrence of the string `kde5` with the string `kde`?

> People are confused by that and try to find a non-existing kde5.

When I originally added the KDE Frameworks, the KDE Plasma 5, and the KDE Applications packages, tried to have "kde.frameworks5" and "kde.plasma5", and our users complained, "Why don't we have KDE 5 yet? Where is KDE 5? How do I install KDE 5?"

Let me politely submit to you that there is a branding problem here, and it's not on our end.

> kde.frameworks5, kde.plasma5 and kde.applications (unversioned, it's mixed content)

Mixed content? What is mixed with what? I know you can't mean it's a mix of KDE 4 and 5 components, because KDE 5 doesn't exist. I am confused.

> is fine though; same for the rest of applications ("KDE Applications" is a well defined released product, which does not include stuff like Kile, KMyMoney etc, which have their own independent schedule; I hope that they are handled in a separate namespace).

What you are talking about are not the package names. The package names are exactly the same as upstream: "konsole-5.8.3" and so on. What you are complaining about is our choice of variable identifiers in the Nixpkgs library. I have to ask, would you make the same complaint about our choice of filenames in the Nixpkgs source code? If so, why? and if not, why is this different?

As indicated before, the hierarchy of identifiers is used to ensure that packages are always built with consistent dependencies, so there is a technical reason this is done. But, if you told me there was a trademark issue with our variable names, that would be a different story.

**Comment by tosky** [comment]
On Tuesday, 21 February 2017 11:25:07 CET Thomas Tuegel wrote:
> > Packages and git are different: you don't have translations apart from
> > Frameworks.
> That is irrelevant to our testing process.
Sometimes there are issues related with translations (one was fixed for Plasma 
5.9.1, for example), but ok, others can test this.

> 
> > Also, for naming, please please really switch to the non-versioned
> > namespace if you use kde.
> So I understand correctly, would you be happy if we just replaced every
> occurrence of the string `kde5` with the string `kde`?
> > People are confused by that and try to find a non-existing kde5.
> 
> When I originally added the KDE Frameworks, the KDE Plasma 5, and the KDE
> Applications packages, tried to have "kde.frameworks5" and "kde.plasma5",
> and our users complained, "Why don't we have KDE 5 yet? Where is KDE 5? How
> do I install KDE 5?"
> 
> Let me politely submit to you that there is a branding problem here, and
> it's not on our end.

People can't find KDE5 because, again, KDE5 does not exists. If they are 
looking for the desktop, it's Plasma. And let me politely remind you that 
distributions should help us in this. Basically all the other distributions 
changed the terminology.


> > kde.frameworks5, kde.plasma5 and kde.applications (unversioned, it's mixed
> > content)
> Mixed content? What is mixed with what? I know you can't mean it's a mix of
> KDE 4 and 5 components, because KDE 5 doesn't exist. I am confused.

I never said this. What I mean is that KDE Applications (which you knows as 
you follow the releases) contains both applications that uses kdelibs4 
libraries and applications ported to Frameworks 5 libraries. This will be the 
case for the next 8 months. So it does not make sense to have 
kde5.applications which contains (if I understand it correctly) all the 
content of KDE Applications. Again, what happens with applications not part of 
KDE Applications, like Kile, KMyMoney, Choqok, Krita?

> > is fine though; same for the rest of applications ("KDE Applications" is a
> > well defined released product, which does not include stuff like Kile,
> > KMyMoney etc, which have their own independent schedule; I hope that they
> > are handled in a separate namespace).
> What you are talking about are not the package names. The package names are
> exactly the same as upstream: "konsole-5.8.3" and so on. What you are
> complaining about is our choice of variable identifiers in the Nixpkgs
> library. I have to ask, would you make the same complaint about our choice
> of filenames in the Nixpkgs source code? If so, why? and if not, why is
> this different?

Again, are those variable identifiers visible to the users? Then they are not 
different from, say, groups of packages in Fedora, or meta-packages in Debian-
based distributions (where they all managed to get way with KDE5). If this is 
the case I would say that having users seeing something called "kde5" is 
definitely confusing and it does not help when people come to our channels for 
questions, for example.

> 
> As indicated before, the hierarchy of identifiers is used to ensure that
> packages are always built with consistent dependencies, so there is a
> technical reason this is done. But, if you told me there was a trademark
> issue with our variable names, that would be a different story.

I'm not sure why this continue reference to the trademark (which could be 
indeed an issue, but I think we can settle without going there). Again, if 
this is user visible string related to "kde5", please replace it with 
something else, and I'm sure that it is totally possible.

-- 
Luigi

**Comment by ttuegel** [comment]
> I'm not sure why this continue reference to the trademark (which could be indeed an issue, but I think we can settle without going there). Again, if this is user visible string related to "kde5", please replace it with something else, and I'm sure that it is totally possible.

I'm 100% on your side on this. I wanted to do it this way from the beginning and I was given a variety of reasons why I couldn't, some of which I repeated here under the guise of devil's advocate. A weak rhetorically strategy, I admit, but I think that a strong complaint from upstream that addresses the prior criticisms will be enough to make others see right. (I kept bringing up trademark because a legally-actionable complaint would *force* other members of NixOS to see it this way.)

I don't have time to take care of this today, but I'll see that this happens before we branch off NixOS 17.03 next week.
